runtime/src/serde_snapshot/tests.rs

```diff
@@ -202,6 +202,8 @@ fn test_bank_serialize_style(serde_style: SerdeStyle) {
     bank2.freeze();
     bank2.squash();
     bank2.force_flush_accounts_cache();
+    let mut test_prior_roots = vec![0, 1, 2];
+    bank2.prior_roots.append(&mut test_prior_roots);
```

Comment:
Here I created a test `prior_roots`, and the field will be verified at line 255 `assert!(bank2 == dbank);`.

Reply:
I'm not sure this is the right spot for tests like this. Maybe another test can be added that explicitly is said to test for ~~forwards/~~ backwards compatibility?

Basically I'm not sure what the right behavior is here. An old snapshot should be usable, but will require the Bank to reconstruct `prior_roots` (if that's needed, or is `prior_root` always empty when loading from a snapshot?). A new snapshot won't work with an old validator, so that's not something I think we need to test for.

If I can pinpoint what I'm thinking about and trying to articulate, I'll make sure to reply. Thanks for bearing with me 😊